dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ORCIDHelper.java

```diff
@@ -0,0 +1,127 @@
+import org.orcid.jaxb.model.common.Relationship;
+import org.orcid.jaxb.model.common.WorkType;
+import org.orcid.jaxb.model.v3.release.common.CreatedDate;
+import org.orcid.jaxb.model.v3.release.common.LastModifiedDate;
+import org.orcid.jaxb.model.v3.release.common.Subtitle;
+import org.orcid.jaxb.model.v3.release.common.Title;
+import org.orcid.jaxb.model.v3.release.common.Url;
+import org.orcid.jaxb.model.v3.release.record.ExternalID;
+import org.orcid.jaxb.model.v3.release.record.ExternalIDs;
+import org.orcid.jaxb.model.v3.release.record.Work;
+import org.orcid.jaxb.model.v3.release.record.WorkTitle;
+
+public final class ORCIDHelper {
+    // TODO: Change to non-sandbox for Dockstore use while using sandbox for test
+    private static final String ORCID_SANDBOX_BASE_URL = "https://api.sandbox.orcid.org/v3.0/";
+    private static final String ORCID_BASE_URL = "https://api.orcid.org/v3.0/";
+
+    private ORCIDHelper() {
+    }
+
+    /**
+     * Construct the XML for an ORCID work so that it can be posted using the ORCID API
+     * Current populated fields are Title, Subtitle, Last Modified, CreatedDate, DOI URL, DOI value, Short description
+     * External ID value must be unique, everything else can be the same (title, subtitle, etc)
+     * An entry (and an optional version) in Dockstore can sent to ORCID
+     * A DOI for the entry or version is required
+     * @param e The entry to be sent to ORCID
+     * @param optionalVersion   Optional version of the entry to send to ORCID
+     * @return An ORCID Work to be sent to ORCID
+     * @throws JAXBException
+     * @throws DatatypeConfigurationException
+     */
+    public static String getOrcidWorkString(Entry e, Optional<Version> optionalVersion) throws JAXBException, DatatypeConfigurationException {
+        Work work = new Work();
+        WorkTitle workTitle = new WorkTitle();
+        Title title = new Title();
+        Subtitle subtitle = new Subtitle("Dockstore Workflow");
+        ExternalIDs externalIDs = new ExternalIDs();
+        ExternalID externalID = new ExternalID();
+        externalID.setType("doi");
+        if (optionalVersion.isPresent()) {
+            Version v = optionalVersion.get();
+            Url url = new Url(v.getDoiURL());
+            externalID.setUrl(url);
+            externalID.setValue(v.getDoiURL());
+            title.setContent(e.getEntryPath() + ":" + v.getName());
+        } else {
+            Url url = new Url(e.getConceptDoi());
+            externalID.setUrl(url);
+            externalID.setValue(e.getConceptDoi());
+            title.setContent(e.getEntryPath());
+        }
+        Title journalTitle = new Title();
+        journalTitle.setContent("Dockstore");
+        work.setJournalTitle(journalTitle);
+        workTitle.setTitle(title);
+        workTitle.setSubtitle(subtitle);
+        work.setWorkTitle(workTitle);
+        work.setShortDescription("A workflow exported from Dockstore");
```

Comment:
Not sure how much of this sort of feedback you want at this point, but ideally it would be the actual workflow name or something like that.

Reply:
The actual workflow name is already in the title. Repeating it in the description seems redundant. I was originally considering just leaving this empty. Can you explain your reasoning on why you want it to be the actual workflow name?

Reply:
I see these 3 strings in this code:

* Dockstore workflow
* Dockstore
* A workflow export from Dockstore

I didn't look closely at the relationship between the 3, but I imagine if I look at several workflows in my ORCID profile, I would keep seeing the same 3 strings again and again, not being able to tell them apart. So I picked on the of the 3 strings to comment on.

Reply:
The PR comment mentions the resulting work in my profile.  It may be useful to look at that. short description is supposed to be well... a short description so it should definitely not be just a workflow name.  I'll keep it as the current one unless I hear any better options. "Dockstore" is the journal title which parallels something like "Nature" in https://orcid.org/0000-0002-6130-1021 .  The "Dockstore workflow" subtitle is mostly redundent I suppose.

Reply:
Hadn't clicked on the ORCID link, and that does help.

I feel we might want to revisit the title at some point as it is in your profile. It doesn't read as nicely as the elements in Denis' profile. But I approved the PR, so we can revisit later.

Reply:
Short description might be the workflow description, agreed not the workflow name. Note that some papers are using the abstract for the description ("Large-Scale Uniform Analysis of Cancer Whole Genomes in Multiple Computing Environments") in my profile. 
Meh on subtitle , agreed that the "Dockstore" is the "journal title"

One other prototype thing ahead of hookup would be to make sure that PUT modification from a newer workflow version works the way we expect (should be different PR, create ticket)

Reply:
I agree the workflow description shouldn't be the workflow name. I also think the subtitle doesn't makes sense and most of the examples in the link you gave don't use it at all. The one that does use the subtitle is a one liner description of the actual work. 

The descriptions we capture have such varying lengths that some make sense as a subtitle, some the short description, and some are probably too long for both. But I think just making them the short description makes the most sense for now.